mars/services/task/analyzer/analyzer.py

```diff
@@ -315,19 +315,27 @@ def _gen_subtask_info(
             extra_config=self._extra_config,
         )
 
-        if (
-            self._has_shuffle
-            and self._shuffle_fetch_type == ShuffleFetchType.FETCH_BY_INDEX
-        ):
-            shuffle_chunks = [
-                c for c in result_chunks if isinstance(c, MapReduceOperand)
-            ]
-            # ensure no shuffle mapper chunks fused into same subtask and subtask only produce mapper outputs
-            # which can be merged dynamically by the reducers.
-            assert len(shuffle_chunks) <= 1, shuffle_chunks
-            proxy_chunks = [c for c in chunks if isinstance(c.op, ShuffleProxy)]
-            assert len(proxy_chunks) <= 1, proxy_chunks
-        return subtask, inp_subtasks
+        is_shuffle_proxy = False
+        if self._has_shuffle:
+            mapper_chunks, proxy_chunks = [], []
+            for c in result_chunks:
+                if (
```

Comment:
This logic is only for the `ShuffleFetchType.FETCH_BY_INDEX`? Record the shuffle proxy subtask reduces duplicate search shuffle proxy overhead, but introduces overhead for `ShuffleFetchType.FETCH_BY_KEY`.

Reply:
It's used by `TaskStageProcessor`